files/lib/gluon/ssid-changer/ssid-changer.sh

```diff
@@ -6,44 +6,44 @@ ONLINE_SSID=$(uci get wireless.client_radio0.ssid -q)
 : ${ONLINE_SSID:=FREIFUNK}   # if for whatever reason ONLINE_SSID is NULL
 OFFLINE_PREFIX='FF_OFFLINE_' # Use something short to leave space for the nodename
 
-UPPER_LIMIT='55' #Above this limit the online SSID will be used
-LOWER_LIMIT='45' #Below this limit the offline SSID will be used
+UPPER_LIMIT='55' # Above this limit the online SSID will be used
+LOWER_LIMIT='45' # Below this limit the offline SSID will be used
 # In-between these two values the SSID will never be changed to preven it from toggeling every Minute.
 
 # Generate an Offline SSID with the first and last Part of the nodename to allow owner to recognise wich node is down
 NODENAME=`uname -n`
-if [ ${#NODENAME} -gt $((30 - ${#OFFLINE_PREFIX})) ] ; then #32 would be possible as well
-	HALF=$(( (28 - ${#OFFLINE_PREFIX} ) / 2 )) #calculate the length of the first part of the node identifier in the offline-ssid
-	SKIP=$(( ${#NODENAME} - $HALF )) #jump to this charakter for the last part of the name
+if [ ${#NODENAME} -gt $((30 - ${#OFFLINE_PREFIX})) ] ; then # 32 would be possible as well
+	HALF=$(( (28 - ${#OFFLINE_PREFIX} ) / 2 )) # calculate the length of the first part of the node identifier in the offline-ssid
+	SKIP=$(( ${#NODENAME} - $HALF )) # jump to this charakter for the last part of the name
 	OFFLINE_SSID=$OFFLINE_PREFIX${NODENAME:0:$HALF}...${NODENAME:$SKIP:${#NODENAME}} # use the first and last part of the nodename for nodes with long name
 else
-	OFFLINE_SSID="$OFFLINE_PREFIX$NODENAME" #greate we are able to use the full nodename in the offline ssid
+	OFFLINE_SSID="$OFFLINE_PREFIX$NODENAME" # great! we are able to use the full nodename in the offline ssid
 fi
 
-#Is there an active Gateway?
-GATEWAY_TQ=`batctl gwl | grep "^=>" | awk -F'[()]' '{print $2}'| tr -d " "` #Grep the Connection Quality of the Gateway which is currently used
+#is there an active gateway?
+GATEWAY_TQ=$(batctl gwl | grep "^=>" | awk -F '[()]' '{print $2}' | tr -d " ") # grep the connection quality of the currently used gateway
 
-if [ ! $GATEWAY_TQ ]; #If there is no gateway there will be errors in the following if clauses
+if [ ! $GATEWAY_TQ ]; # if there is no gateway there will be errors in the following if clauses
 then
-	GATEWAY_TQ=0 #Just an easy way to get an valid value if there is no gatway
+	GATEWAY_TQ=0 # just an easy way to get a valid value if there is no gateway
 fi
 
 if [ $GATEWAY_TQ -gt $UPPER_LIMIT ];
 then
 	echo "Gateway TQ is $GATEWAY_TQ node is online"
-	for HOSTAPD in $(ls /var/run/hostapd-phy*); do #Check status for all physical devices
+	for HOSTAPD in $(ls /var/run/hostapd-phy*); do # check status for all physical devices
 		CURRENT_SSID=`grep "^ssid=$ONLINE_SSID" $HOSTAPD | cut -d"=" -f2`
-		if [ $CURRENT_SSID == $ONLINE_SSID ]
+		if [ "X$CURRENT_SSID" == "X$ONLINE_SSID" ]
```

Comment:
`if [ "$CURRENT_SSID" == "$ONLINE_SSID" ]` should be sufficient for casting to string. There's no need to prepend an `X`.